trin-core/src/portalnet/overlay_service.rs

```diff
@@ -848,13 +965,43 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
             self.protocol,
             source.node_id()
         );
-        self.process_discovered_enrs(
-            nodes
-                .enrs
-                .into_iter()
-                .map(|ssz_enr| ssz_enr.into())
-                .collect(),
-        );
+
+        let enrs: Vec<Enr> = nodes
+            .enrs
+            .into_iter()
+            .map(|ssz_enr| ssz_enr.into())
+            .collect();
+
+        let enrs_clone = enrs.clone();
+        self.process_discovered_enrs(enrs);
+
+        let node_id = source.node_id();
+
+        // if this is part of a query, update the query
+        if let Some(query_id) = self.active_query_requests.write().remove(&node_id) {
```

Comment:
This logic would be common to both `Nodes` and possibly `Content` responses, so we could put this logic into `process_discovered_enrs`.